UIComponets/ViewController/TextFiledDatePickerVC.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  TextFiledDatePickerVC.swift
+//  UIComponets
+//
+//  Created by Rajan Patel on 26/05/23.
+//
+
+import UIKit
+
+class TextFiledDatePickerVC: UIViewController {
+
+    // MARK: - IBOutlets
+    @IBOutlet weak var txtfDate: UITextField!
+    
+    // MARK: - Variables And Declarations
+    let datePicker = UIDatePicker()
+    
+    // MARK: - Life cycle Methods
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setViews()
+    }
+    
+    // MARK: - Set up
+    private func setViews() {
+        datePicker.datePickerMode = .dateAndTime
+        datePicker.preferredDatePickerStyle = .inline
+        datePicker.addTarget(self, action: #selector(datePickerValueChange), for: .valueChanged)
+        txtfDate.inputView = datePicker
+        txtfDate.delegate = self
+        
+        let toolbar = UIToolbar()
+        toolbar.sizeToFit()
+        
+        let btnDone = UIBarButtonItem(title: "Done", style: .done, target: self, action: #selector(btnDoneClicked))
+        let flaxibleSpace = UIBarButtonItem(barButtonSystemItem: .flexibleSpace, target: nil, action: nil)
+        toolbar.setItems([flaxibleSpace, btnDone], animated: true)
+        
+        txtfDate.inputAccessoryView = toolbar
+    }
+    
+    // MARK: - Action
+    @objc func btnDoneClicked(button: UIButton) {
+        txtfDate.resignFirstResponder()
+    }
+    
+    // MARK: - Action
+    @objc func datePickerValueChange(_ sender: UIDatePicker) {
```

Comment:
This method occures after value changes so its should be in past tent _datePickerValueChanged_ , if you observer any build in menthod on apple it focuse on tenses, such as scrollViewwillScroll(), viewDidApper()